### PR TITLE
Refactor viewport-modify to resize-rectangle-interaction

### PR DESCRIPTION
**Pull request by Dassderdie**
This PR replaces the very hacky implementation to resize the viewport with a new one.

* This can be merged after #601 
* Instead of using some parts of the `Modify` interaction (side-effect in the style function) a new custom Interaction has been created.
* Previously, one could switch between resizing and translating a viewport via pressing `Shift`. Instead, now one ***resizes*** if one drags an edge of the viewport (+ a bit around it), and ***translates*** if one drags any other part. This should enable touch users to resize viewports (I haven't tested it yet).
* Regression: Currently, there is no indicator that one can resize or is currently resizing a viewport. Previously this was done via the blue circle of the Modify interaction. If wanted, this could later be added either by modifying the style of the viewport or adding an overlay to the `ResizeRectangleInteraction`.
* The code should now be more readable and extendable.
* The names have been modified so that this interaction is now not specific for a viewport and easier to reuse.
* The unnecessary `viewportTranslateInteraction` has been removed
* `feature.dispose()` now gets called when it gets deleted - this could have been a memory leak?

### CODE REVIEW

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/viewport-feature-manager.ts [hunk 1]
```diff
@@ -61,15 +61,9 @@ export class ViewportFeatureManager
...
-    private readonly modifyHelper = new ModifyHelper();
 
     private readonly style = new Style({
-        geometry(thisFeature) {
-            const modifyGeometry = thisFeature.get('modifyGeometry');
-            return modifyGeometry
-                ? modifyGeometry.geometry
-                : thisFeature.getGeometry();
-        },
+        fill: undefined,
```
Comment (Dassderdie) [inline frontend/src/app/pages/exercises/exercise/shared/exercise-map/feature-managers/viewport-feature-manager.ts:66]:
```suggestion
```

I worked on a branche where #603 was already partially implemented -> this should not be changed, yet

**Comment by Greenscreen23** [comment]
We copied your code for `viewports` to `simulatedRegions` because of several reasons. Do you mind applying this to `simulatedRegions` as well when they are merged?

**Comment by Dassderdie** [comment]
> We copied your code for `viewports` to `simulatedRegions` because of several reasons. 

Do you mean #601 (as of yesterday)? Or are you currently copying something from the viewports of this PR?

> Do you mind applying this to `simulatedRegions` as well when they are merged?

This was the plan.

**Comment by Greenscreen23** [comment]
> Do you mean https://github.com/hpi-sam/digital-fuesim-manv/pull/601 (as of yesterday)? Or are you currently copying something from the viewports of this PR?
As of yesterday, I've deduplicated it today.

**Review by Greenscreen23 — APPROVED** [review]
It integrates properly with our setup and works as expected. I've taken a look over the code, but I'm trusting you that this code will work fine with ol, as I don't think I'm experienced enough to understand ol.

File: frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/resize-rectangle-interaction.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    originCorner: Coordinate;
+    /**
+     * The feature that is currently being resized.
+     */
+    feature: Feature<LineString>;
+    currentScale: { x: number; y: number };
+}
+
+// TODO: This doesn't work as a static member of ResizeEvent, because it is undefined at runtime. Why?
+const resizeRectangleEventType = 'resizerectangle';
```
Comment (ClFeSc) [inline frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/resize-rectangle-interaction.ts:140]: No idea. My only guess would be that something with a conflicting name already exists in the type hierarchy.

**Review by ClFeSc — APPROVED** [review]
I haven't tested it, but the code looks good. Much cleaner than before.

**Comment by ClFeSc** [comment]
@Dassderdie Regarding the question of what we used last year, it might have been this article: <https://openlayers.org/en/latest/examples/modify-scale-and-rotate.html>.

**Comment by Greenscreen23** [comment]
:shipit:

**Pull request merged**